### PR TITLE
Added logic to further support observations in HQMF+CQL

### PR DESCRIPTION
**Pull request by holmesie**
This is the third of three pull requests, the other two are:
1. https://github.com/projectcypress/health-data-standards/pull/506
2. https://github.com/projecttacoma/bonnie_bundler/pull/87

before:
1_after

after:

before:

after:
2_after

Measures to test with:
[CMS55_v5_1_Artifacts.10.zip](https://github.com/projecttacoma/bonnie/files/1108862/CMS55_v5_1_Artifacts.10.zip)
[Test32_v5_2_Artifacts.zip](https://github.com/projecttacoma/bonnie/files/1135954/Test32_v5_2_Artifacts.zip)

Related internal JIRA test plans:
https://jira.mitre.org/browse/BONNIE-752
https://jira.mitre.org/browse/BONNIE-754

### CODE REVIEW

File: app/assets/javascripts/cql_calculator.js.coffee [hunk 1]
```diff
@@ -151,6 +153,14 @@
...
           else
             population_results[popCode] = 0
+      else if popCode == 'OBSERV' && observation_defs?.length > 0
+        # Handle observations using the names of the define statements that
+        # were added to the ELM to call the observation functions.
+        for ob_def in observation_defs
+          population_results['values'] = [] unless population_results['values']
+          obs_result = results['patientResults']?[patient.id]?[ob_def]
+          if obs_result && obs_result.length > 0
+            population_results['values'].push(obs_result[0].value)
```
Comment (edeyoung) [inline app/assets/javascripts/cql_calculator.js.coffee]: note assumption of a single valued results. (is this assumption accurate? how do we know?)

File: app/assets/javascripts/cql_calculator.js.coffee [hunk 2]
```diff
@@ -83,9 +83,11 @@
           main_library_index = index
 
       observations = population.collection.parent.get('observations')
+      observation_defs = []
       if observations
          for obs in observations
-           generatedELMJSON =  @generateELMJSONFunction(obs.function_name, obs.parameter)
+           generatedELMJSON = @generateELMJSONFunction(obs.function_name, obs.parameter)
+           observation_defs.push('BonnieFunction_' + obs.function_name)
```
Comment (edeyoung) [inline app/assets/javascripts/cql_calculator.js.coffee]: "BonnieFunction" should probably be a constant defined somewhere... but it's not. so add a comment explaining that this is what gets added to the function definition name to create the definition name.

**Review by edeyoung — CHANGES_REQUESTED** [review]

File: app/assets/javascripts/cql_calculator.js.coffee (hunk 1 above)
Comment (holmesie) [reply app/assets/javascripts/cql_calculator.js.coffee]: Yeah, check the HQMF DTSU spec for CQL (in Paul's share), look at section 4.29 	measureObservationDefinition

File: app/assets/javascripts/cql_calculator.js.coffee (hunk 2 above)
Comment (holmesie) [reply app/assets/javascripts/cql_calculator.js.coffee]: Done!

**Review by mayerm94 — APPROVED** [review]
I have tested since your last changes, everything still works as expected

**Comment by edeyoung** [comment]
Please test with re-uploading the attached measure on hmdmlantana. Note that the stratifications are in the population context not the patient context (i've asked her to revise and re-upload). However, the first population should still work. With this code and re-uploading that measure, I do not see the OBSERV results.

[Test32_v5_1_Artifacts.zip](https://github.com/projecttacoma/bonnie/files/1127630/Test32_v5_1_Artifacts.zip)

**Review by edeyoung — APPROVED** [review]

**Comment by edeyoung** [comment]
HDS and Bonnie-Bundler have been merged. Gemfile.lock needs to be updated to point to these branches.

**Comment by edeyoung** [comment]
Also, looks like it needs to be updated with the latest.

**Review by jbradl11 — APPROVED** [review]

**Pull request merged**